src/app/service/http.service.ts

```diff
@@ -1,9 +1,23 @@
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 
 @Injectable({
   providedIn: 'root'
 })
 export class HttpService {
 
-  constructor() { }
+  constructor(private http: HttpClient) { }
+
+  
+  getJokes(){
+    return this.http.get('https://api.chucknorris.io/jokes/random')
+  }
+
+  getCategories(){
+    return this.http.get('https://api.chucknorris.io/jokes/categories')
+  }
+
+  categoryExists(){
+    return this.http.get('https://api.chucknorris.io/jokes/random?category={category}')
```

Comment:
usually, when something's in curly brackets it means it's a variable, so it should be replaced with actual value for example if selected category is an animal, {category} should be changed with animal